build.gradle

```diff
@@ -834,6 +839,21 @@ project(':iceberg-snowflake') {
     runtimeOnly("net.snowflake:snowflake-jdbc")
 
     testImplementation project(path: ':iceberg-core', configuration: 'testArtifacts')
+    integrationImplementation("net.snowflake:snowflake-jdbc")
+  }
+
+  sourceSets {
+    integration {
+      java.srcDir "$projectDir/src/integration/java"
+      resources.srcDir "$projectDir/src/integration/resources"
+      compileClasspath += main.output + test.output
+      runtimeClasspath += main.output + test.output
+    }
+  }
+
+  task integrationTest(type: Test) {
+    testClassesDirs = sourceSets.integration.output.classesDirs
+    classpath = sourceSets.integration.runtimeClasspath
```

Comment:
this also requires `check.dependsOn integrationTest` below L857, otherwise integration tests are never running as part of CI

Reply:
The decision to not run integrationTest as part of CI is intentional until we reach milestone where snowflake side APIs/Syntax are documented and stable. In current state, the integration tests are meant to be run by the catalog developers who have access to Private preview.